src/toil/leader.py

```diff
@@ -827,7 +828,9 @@ def processFinishedJob(self, batchSystemID, resultStatus, wallTime=None):
                             else:
                                 logger.warn('The batch system left an empty file %s' % batchSystemFile)
 
-                jobGraph.setupJobAfterFailure(self.config)
+                # Do not decrement retry count for lost jobs (i.e. due to preemptable failures).
+                decrementRetryCount = not self.config.enableUnlimitedPreemptableRetries or resultStatus != LOST_JOB_EXIT_STATUS
```

Comment:
I think this check should live in `setupJobAfterFailure`. It makes more sense to pass it the failure reason for the job than a single-purpose flag, I think, and it already has access to the config.